standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1408,6 +1410,43 @@ private boolean isUpdateOrDelete(Statement stmt, String conflictSQLSuffix) throw
     }
   }
 
+  public long getLatestTxnInConflict(long txnid) throws MetaException {
```

Comment:
Maybe add some javadoc, what does it do. I would emphasise that this method call only makes sense if the caller holds an exclusive lock, that blocks other txns to commit writes. And also, that this deliberately ignores inserts, or maybe this should be added to the DriverTxnHandler, that inserts will not invalidate the snapshot, that can cause duplicates

Reply:
added javadoc